src/runtime_src/doc/toc/xrt_kernel_executions.rst

```diff
@@ -0,0 +1,105 @@
+
+Mode of operation
+-----------------
+
+.. image:: ap_ctrl_hs_1.png
+   :align: center
+
+1.	The XRT driver writes a 1 in ap_start to start the kernel 
+2.	The XRT driver waits for ap_done to be asserted by the kernel (guaranteeing the output data is fully produced by the kernel). 
+3.	Repeat 1-2 for next kernel execution 
+
+Assume there are three concurrent kernel execution requests from the host. The kernel executions will happen sequentially as below, serving one request at a time
+
+|START1|=>|DONE1|=>|START2|=>|DONE2|=>|START3|=>|DONE3|
+
+Control Signal Topology
+-----------------------
+The signals ap_start and ap_done must be connected to the AXI_LITE control and status register (at the address 0x0 of the AXI4-Lite Slave interface) section to specific bits.  
+
+====== ===================== =======================================================================
+ Bit       Signal name                                  Description
+====== ===================== =======================================================================
+  0         ap_start           Asserted by the XRT when kernel can process the new data
+  1         ap_done            Asserted by the kernel when it is finished producing the output data
+====== ===================== =======================================================================
+
+================================
+AP_CTRL_CHAIN (Pipelined Kernel)
+================================
+
+AP_CTRL_CHAIN is an improved version of AP_CTRL_HS where the kernel is designed in such a way it can allow multiple kernel executions to get overlapped and running in a pipelined fashion. To achieve this host to kernel synchronization point is broken into two places: input synchronization (dictated by the signals **ap_start** and **ap_ready**) and output synchronization (**ap_done** and **ap_continue**). This execution mode allows the kernel to be restarted even if the kernel is working on the current (one or more) execution(s). So when there are multiple kernel execution requests from the host, the kernel gets executed in a pipelined or overlapping fashion, serving multiple execution requests at a time. 
+
+Mode of operation
+-----------------
+
+.. image:: ap_ctrl_chain_1.png
+   :align: center
+
+**Input synchronization**
+
+1.	The XRT driver writes a 1 in ap_start to start the kernel 
+2.	The XRT driver waits for ap_ready to be asserted by the kernel (guaranteeing the kernel is ready to accept new data for next execution, even if it is still working on the previous execution request). 
+3.	The XRT driver writes 1 in ap_start to start the kernel operation again
+
+Assume there are five concurrent kernel execution requests from the host and the kernel can work on three execution requests in a pipelined fashion. The kernel executions will happen sequentially as below, serving maximum three requests at a time. 
+
+|START1|=>|START2|=>|START3|=>|DONE1|=>|START4|=>|DONE2|=>|START5|=>|DONE3|=>|DONE4|=>|DONE5|
+
+
```

Comment:
We should call emphasize that AP_CTRL_CHAIN implies all requests are serviced _in order_.

Reply:
Good point, added the following:
|START1|=>|START2|=>|START3|=>|DONE1|=>|START4|=>|DONE2|=>|START5|=>|DONE3|=>|DONE4|=>|DONE5|

**Note:** As noted in the above sequence, the AP_CTRL_CHAIN only applicable when the kernel produces the outputs for the pending requests in-order. Kernel servicing the requests out-of-order cannot be supported by AP_CTRL_CHAIN execution model.